sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java

```diff
@@ -668,6 +672,74 @@ public void testReadMessagesWithCoderAndParseFn() {
 
     List<String> outputs = ImmutableList.of("foo", "bar");
     PAssert.that(read).containsInAnyOrder(outputs);
-    readPipeline.run();
+    pipeline.run();
+  }
+
+  @Test
+  public void testDynamicTopicsBounded() throws IOException {
+    testDynamicTopics(true);
+  }
+
+  @Test
+  public void testDynamicTopicsUnbounded() throws IOException {
+    testDynamicTopics(false);
+  }
+
+  public void testDynamicTopics(boolean isBounded) throws IOException {
+    List<OutgoingMessage> outgoing =
```

Comment:
nit: name expectedOutgoing, it wasn't obvious that the test factory for publish took these as expectations when created. I think naming would help.

Reply:
done